iothub/device/src/ConnectionInfo.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// The connection information since the last state change.
+    /// </summary>
+    public class ConnectionInfo
+    {
+        internal ConnectionInfo() { }
+
+        internal ConnectionInfo(ConnectionState state, ConnectionStateChangeReason changeReason, DateTimeOffset changedOnUtc)
+        { 
+            State = state;
+            ChangeReason = changeReason;
+            ChangedOnUtc = changedOnUtc;
+        }
+
+        /// <summary>
+        /// The current connection state of the device or module.
+        /// </summary>
+        /// <remark>
+        /// Defaults to <see cref="ConnectionState.Disconnected"/>.
+        /// </remark>>
+        public ConnectionState State { get; internal set; } = ConnectionState.Disconnected;
```

Comment:
Can these be readonly now?

```suggestion
        public ConnectionState State { get; } = ConnectionState.Disconnected;
```